web/src/layout/map.js

```diff
@@ -46,18 +46,23 @@ export default () => {
 
   const [tooltipPosition, setTooltipPosition] = useState(null);
   const [tooltipZoneData, setTooltipZoneData] = useState(null);
+  const [hasCentered, setHasCentered] = useState(false);
+
+  // Center the map initially based on the focused zone and the user geolocation.
+  if (!hasCentered) {
+    if (zoneId) {
+      console.log(`Centering on zone ${zoneId}`);
+      dispatchApplication('mapViewport', getCenteredZoneViewport(zones[zoneId]));
+      setHasCentered(true);
+    } else if (callerLocation) {
+      console.log(`Centering on browser location (${callerLocation})`);
+      dispatchApplication('mapViewport', getCenteredLocationViewport(callerLocation));
+      setHasCentered(true);
+    }
+  }
```

Comment:
Hmm, shouldn't this ideally happen in a `useEffect()` block with `hasCentered` as dependency instead? 🤔

Reply:
of course you're right 🤦